cypress-tests/cypress/support/redirectionHandler.js

```diff
@@ -412,30 +412,23 @@ function verifyReturnUrl(redirection_url, expected_url, forward_flow) {
   const urlParams = new URLSearchParams(redirection_url.search);
   const paymentStatus = urlParams.get('status');
 
-  // Check for valid statuses
-  if (statusCode >= 400 && statusCode < 500) {
-    console.error(`Client Error: ${statusCode} - ${paymentStatus || "Payment status not available"}`);
-    throw new ClientError(`Client error occurred. Status code: ${statusCode}`);
-  } else if (statusCode >= 500) {
-    console.error(`Server Error: ${statusCode} - ${paymentStatus || "Payment status not available"}`);
-    throw new ServerError(`Server error occurred. Status code: ${statusCode}`);
-  } else if (!paymentStatus) {
-    console.error(`Error: Payment status is undefined. Unexpected error.`);
-    throw new Error(`Payment status is undefined. Unexpected error.`);
+  if (!paymentStatus) {
+    console.error(`Error: Payment status is undefined. This might indicate a 4xx or 5xx error.`);
+    throw new Error(`Payment status is undefined. There may have been a client or server error (4xx/5xx). Check network logs or UI.`);
```

Comment:
would be great if you could make it take a screenshot when redirection fails.

to do so, you can use `cy.screenshot()` command.